django/db/models/sql/compiler.py

```diff
@@ -1847,7 +1850,6 @@ def execute_sql(self, returning_fields=None):
                         ),
                     )
                 ]
-        cols = [field.get_col(opts.db_table) for field in self.returning_fields]
```

Comment:
I would prefer if we define a default `cols` before the `if` and override its value in the specific guard where something else is needed for cols. Adding a variable to `locals` only inside `if` branches, as opposed to defining it before the `if` statement, can lead to potential issues and is generally considered less readable and more error-prone. When you define a variable inside an `if` block, its theoretical scope is limited to that block.

Future modifications could lead to `NameError: name 'cols' is not defined` because someone missed that `cols` should be defined in any new branch of the `if`.

Reply:
Sure, I added `cols = []`.